src/vs/base/browser/ui/button/button.ts

```diff
@@ -169,12 +174,13 @@ export class Button extends Disposable implements IButton {
 
 	private applyStyles(): void {
 		if (this._element) {
-			let background, foreground;
+			let background, foreground, separator;
 			if (this.options.secondary) {
 				foreground = this.buttonSecondaryForeground ? this.buttonSecondaryForeground.toString() : '';
 				background = this.buttonSecondaryBackground ? this.buttonSecondaryBackground.toString() : '';
 			} else {
 				foreground = this.buttonForeground ? this.buttonForeground.toString() : '';
+				separator = this.buttonSeparator ? this.buttonSeparator.toString() : '';
```

Comment:
This is one part where I wasn't sure if this needed to be part of this condition since there is no separator in the secondary button. Happy to move this outward.

Reply:
I think we have to take it out given that currently it breaks compilation.

Reply:
Ok pushed changes to fix the compilation errors